web/html/src/manager/proxy/container-config.tsx

```diff
@@ -0,0 +1,359 @@
+import * as React from "react";
+
+import { AsyncButton } from "components/buttons";
+import { SubmitButton } from "components/buttons";
+import { Form } from "components/input/Form";
+import { unflattenModel } from "components/input/form-utils";
+import { FormMultiInput } from "components/input/FormMultiInput";
+import { Radio } from "components/input/Radio";
+import { Text } from "components/input/Text";
+import { Panel } from "components/panels/Panel";
+import { TopPanel } from "components/panels/TopPanel";
+import Validation from "components/validation";
+
+import { DEPRECATED_unsafeEquals } from "utils/legacy";
+import Network from "utils/network";
+
+import { ContainerConfigMessages } from "./container-config-messages";
+
+enum SSLMode {
+  UseSSL = "use-ssl",
+  CreateSSL = "create-ssl",
+}
+
+const initialModel = {
+  caCertificate: "",
+  caKey: "",
+  caPassword: "",
+  country: "",
+  state: "",
+  city: "",
+  org: "",
+  orgUnit: "",
+  sslEmail: "",
+  rootCA: "",
+  proxyCertificate: "",
+  proxyKey: "",
+  proxyAdminEmail: "",
+  sslMode: SSLMode.CreateSSL,
+  maxSquidCacheSize: "",
+  proxyFQDN: "",
+  serverFQDN: "",
+  proxyPort: "8022",
+};
+
+function replaceCRLF(content: string | undefined | null) {
+  return content != null ? content.replace(/\r\n/g, "\n") : content;
+}
+
+export function ProxyConfig() {
+  const [messages, setMessages] = React.useState<string[]>([]);
+  const [loading, setLoading] = React.useState(false);
+  const [success, setSuccess] = React.useState<boolean | undefined>();
+  const [model, setModel] = React.useState(initialModel);
+  const [invalid, setInvalid] = React.useState(false);
+
+  const onSubmit = () => {
+    setMessages([]);
+    setLoading(true);
+
+    const fileFields = {
+      caCertificate: SSLMode.CreateSSL,
+      caKey: SSLMode.CreateSSL,
+      rootCA: SSLMode.UseSSL,
+      intermediateCAs: SSLMode.UseSSL,
+      proxyCertificate: SSLMode.UseSSL,
+      proxyKey: SSLMode.UseSSL,
+    };
+
+    const fileReaders = Object.keys(model)
+      .filter((key) => {
+        const matcher = key.match(/^([a-zA-Z0-9]*[A-zA-Z])[0-9]+$/);
+        const fieldName = matcher ? matcher[1] : key;
+        return fileFields[fieldName] === model.sslMode;
+      })
+      .map((fieldName) => {
+        const field = document.getElementById(fieldName);
+        if (field != null && field instanceof HTMLInputElement && field.files != null) {
```

Comment:
Where possible, I would recommend being specific about whether you check against `null`, `undefined` or both. For example, `document.getElementById` will never return undefined, so it would be better to check `if (field !== null ...)`, whereas below the `Promise` will never be `null`, so it would be better to check against `undefined` instead.  

The reason I recommend this is for clarity. If I expect something to be a value or null but then suddenly undefined also pops up from somewhere, I will start to look for where I missed that case and vice versa.